homeassistant/components/hdmi_cec/__init__.py

```diff
@@ -374,10 +374,31 @@ class CecEntity(Entity):
     def __init__(self, device, logical) -> None:
         """Initialize the device."""
         self._device = device
-        self._icon = None
         self._state: str | None = None
         self._logical_address = logical
         self.entity_id = "%s.%d" % (DOMAIN, self._logical_address)
+        self._attr_should_poll = False
+        self._set_attr_name()
+        self._attr_icon = (
+            ICONS_BY_TYPE.get(self._device.type)
+            if self._device.type in ICONS_BY_TYPE
+            else ICON_UNKNOWN
+        )
+
+    def _set_attr_name(self):
```

Comment:
? Why do this and not use the previous property method?

Reply:
Because the goal of this PR is to use _attr_variable instead of the indirect way via @property. I had it in the init method, but you did not like the complexity it added.

Reply:
That comment makes no sense, you have just replaced one method with another. Why would this be better than the old/previous code?

Reply:
My preference was to have it in the init method, which you did not like. 

having a method only called from init and a @property method is 2 different pair of shoes.
Using @property means the method is called every time core wants to know the name, while the method now is only called from init.

But I am more than happy to move it back into init where it belongs (in my mind).

Reply:
I think we should prevent large logic from ending up in the constructor as well.

@MartinHjelmare What do you think we should do in these cases?

Reply:
I think it's ok to call a method to set the instance attributes, as long as there are no side effects.